src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -413,21 +410,31 @@ export class ChallengesubmitComponent implements OnInit {
 
     const FORM_DATA: FormData = new FormData();
     FORM_DATA.append('status', 'submitting');
-    FORM_DATA.append('input_file', self.globalService.formItemForLabel(self.components, 'input_file').fileSelected);
+    if (!self.isSubmissionUsingUrl) {
+      FORM_DATA.append('input_file', self.globalService.formItemForLabel(self.components, 'input_file').fileSelected);
+    }
     FORM_DATA.append('method_name', self.globalService.formValueForLabel(self.components, 'method_name'));
     FORM_DATA.append('method_description', self.globalService.formValueForLabel(self.components, 'method_description'));
     FORM_DATA.append('project_url', self.globalService.formValueForLabel(self.components, 'project_url'));
     FORM_DATA.append('publication_url', self.globalService.formValueForLabel(self.components, 'publication_url'));
+    if (self.validFileUrl === true && self.isSubmissionUsingUrl) {
```

Comment:
How about if you simply write it as `self.validFileUrl && self.isSubmissionUsingUrl`?